src/control/lib/control/system.go

```diff
@@ -139,6 +141,107 @@ func SystemJoin(ctx context.Context, rpcClient UnaryInvoker, req *SystemJoinReq)
 	return resp, convertMSResponse(ur, resp)
 }
 
+// SystemNotifyReq contains the inputs for the system notify request.
+type SystemNotifyReq struct {
+	unaryRequest
+	msRequest
+	Event    events.Event
+	Sequence uint64
+}
+
+// ToClusterEventReq converts the system notify request to a cluster events
+// request. Resolve control address to a hostname if possible.
+func (req *SystemNotifyReq) ToClusterEventReq() (*mgmtpb.ClusterEventReq, error) {
```

Comment:
If this is used only internally, should it be exported?

Reply:
done